crates/pathfinder/src/state/sync/l1.rs

```diff
@@ -1,680 +1,49 @@
-        .when(retry_condition)
-        .await
-}
-
-#[derive(Clone)]
-struct EthereumImpl<T: EthereumTransport + Send + Sync> {
-    logs: Arc<RwLock<StateRootFetcher>>,
-    transport: T,
-}
-
-#[async_trait::async_trait]
-impl<T: EthereumTransport + Send + Sync + Clone> EthereumApi for EthereumImpl<T> {
-    async fn fetch_logs(&mut self) -> Result<Vec<StateUpdateLog>, FetchError> {
-        let ff = || async {
-            let logs = self.logs.clone();
-            let transport = self.transport.clone();
-            let mut logs = logs.write().await;
-            logs.fetch(transport).await
-        };
-        let logs = retry(ff, |error| match error {
-            FetchError::Other(other) => {
-                tracing::warn!(reason=%other, "Failed fetching L1 logs, retrying");
-                true
-            }
-            FetchError::Reorg => false,
-        })
-        .await?;
-        Ok(logs)
-    }
-
-    async fn set_log_head(&mut self, head: Option<StateUpdateLog>) {
-        self.logs.write().await.set_head(head);
-    }
-
-    async fn log_head(&self) -> Option<StateUpdateLog> {
-        self.logs.read().await.head().clone()
-    }
-
-    async fn block_hash(
-        &self,
-        block: EthereumBlockNumber,
-    ) -> anyhow::Result<Option<EthereumBlockHash>> {
-        // No explicit retrying here as any `EthereumTransport` implementor should already hadle that there.
-        Ok(self
-            .transport
-            .block(block.0.into())
-            .await?
-            .map(|b| EthereumBlockHash(b.hash.unwrap().0.into())))
-    }
-}
-
-/// Sends [sync events](Event) on its channel.
-///
-/// Main purpose is to simplify the code in [sync_impl] by abstracting
-/// out the error mapping and oneshot channel receives.
-struct EventSender(mpsc::Sender<Event>);
-
-#[derive(Debug, PartialEq, Clone, Copy)]
-struct ChannelClosedError;
-
-impl EventSender {
-    /// Sends [Event::QueryUpdate] on its channel and returns the result.
-    async fn get_update(
-        &self,
-        block: BlockNumber,
-    ) -> Result<Option<StateUpdateLog>, ChannelClosedError> {
-        let (tx, rx) = oneshot::channel();
-        self.0
-            .send(Event::QueryUpdate(block, tx))
-            .await
-            .map_err(|_send_err| ChannelClosedError)?;
-
-        rx.await.map_err(|_recv_err| ChannelClosedError)
-    }
-
-    /// Sends [Event::Update] on its channel.
-    async fn updates(&self, updates: Vec<StateUpdateLog>) -> Result<(), ChannelClosedError> {
-        self.0
-            .send(Event::Update(updates))
-            .await
-            .map_err(|_send_err| ChannelClosedError)
-    }
-
-    /// Sends [Event::Reorg] on its channel.
-    async fn reorg(&self, block: BlockNumber) -> Result<(), ChannelClosedError> {
-        self.0
-            .send(Event::Reorg(block))
-            .await
-            .map_err(|_send_err| ChannelClosedError)
-    }
+    sync_impl(ethereum, tx_event, chain, core_address).await
```

Comment:
Do we need the extra indirection / abstraction?

Reply:
Done